LiskHQ/lisk-hub
# Implement new design of register a delegate page

### What issue have I solved?

-- #2082 

### How have I implemented/fixed it?

This PR includes the new design for delegate registration.
nickname validation (is already exist or if has valid characters or not).
If user has enough funds or if user is a delegate already.

In the summary page show the fee amount of the tx with the selected nickname and user address.

In the result page or final page after submit the transactions will display the animation based on the result of the tx, in case of success will display a button that will redirect to delegates page, but in case of error a link for report the error and a retry button to submit the tx again.


### How has this been tested?


1. login to the app with an account that is not a delegate already.
2. then go to delegates tab
3. do a click in Become a delegate button
4. type the nickname that you want (you can type anything even special characters and see the validations)
5. Do a click in the confirmation button
6. Check the summary page.
7. Then submit the transactions, (if you want you can click in the in the Go Back button and see the previous page agin)
8. After submit the transaction you will see the animation and once tx is confirmed you will see the success status or not if fail.

**NOTE:** If you select and account with not enough funds you wont be able to do the process.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
